### PR TITLE
Sprint 11

### PR DESCRIPTION
**Pull request by wbyoko**
Court User: Creating a case on behalf of paper-filing petitioner #311

 * Start a new case for a paper-filed Petition

Epic 2 Clean-Up #367
 * Various tasks to correct

* Add Case Caption to Petition Document Detail

* Send Message form / Create New Message

Parties: Assign Doc Type for External Filing #269
 * Parties are able to select a document type from a list of Court-approved documents. 

* Add File Document Button for petitioners, practitioners, and respondent

* Desktop category selection

* Mobile view

### CODE REVIEW

**Comment by waldoj** [comment]
The [code review checklist](https://github.com/ustaxcourt/ef-cms/blob/staging/CODE_REVIEW.md):

- [x] begin by deleting any existing copy of the Git repository and cloning a fresh copy, and running `docker system prune` to remove any cached artifacts from any prior code review
- [ ] review the pull request itself, to get oriented
	- [x] read the description of the pull request, which should summarize the changes made
	- [ ] read through every task on the Scrum board that's encompassed by this pull request
	- [ ] read the description of the commits that comprise the pull request
- [ ] stand up the site locally
	- [ ] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
		- [ ] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
		- [ ] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/), including Internet Explorer 11
		- [ ] test in Mobile Safari and Mobile Chrome, with the caveat that not all internal Court functionality will be necessary on these platforms
	- [ ] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [ ] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [ ] review for accessibility
		- [ ] use an automated audit tool, such as Chrome Audit or aXe
		- [ ] navigate site only with the keyboard
		- [ ] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [ ] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [ ] review static code analysis results in SonarCloud
- [x] run a security audit of dependencies (`npm audit`) in `efcms-service`, `web-client`, and `shared`, to ensure that there are no vulnerabilities that will be deployed to production (as opposed to vulnerabilities that only have an impact on the development environment)
- [x] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [ ] skim all new code, in the context of existing code, [looking for problems](#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [ ] review all tests
	- [ ] look at code coverage of tests in SonarCloud
	- [ ] methodically review all new tests for correctness, quality of naming
- [ ] determine what code isn’t tested, review that rigorously
- [ ] review documentation to ensure that it matches changes
- [ ] provide comments on the pull request on GitHub, as necessary
	- [ ] for comments that are specific to a particular line of code, comment on those specific lines
	- [ ] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments
- [ ] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

**Comment by waldoj** [comment]
I will provide general feedback in the form of line-level feedback to `README.md`.

File: web-client/src/presenter/computeds/formattedCaseDetail.js [hunk 1]
```diff
@@ -110,15 +111,45 @@ const formatCase = (caseDetail, caseDetailErrors) => {
...
+    result.practitioner.formattedName = formattedName;
+  }
 
   result.createdAtFormatted = moment.utc(result.createdAt).format('L');
-  result.irsDateFormatted = moment.utc(result.irsDate).format('L LT');
+  result.receivedAtFormatted = moment.utc(result.receivedAt).format('L');
+  result.irsDateFormatted = moment
+    .utc(result.irsSendDate)
+    .local()
+    .format('L LT');
```
[the remark this answers is not in the document]
Comment (rachaelparris) [reply web-client/src/presenter/computeds/formattedCaseDetail.js]: We made the decision to go with local for now, with the intention of creating a date formatting utility that will do that across the board in the near future.

File: shared/src/tools/generateCategories.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
+const fs = require('fs');
```
[the remark this answers is not in the document]
Comment (rachaelparris) [reply shared/src/tools/generateCategories.js]: I believe we're using the `csv-parse` module to do the actual CSV parsing.

[the remark this answers is not in the document]
Comment (kkoskelin) [reply shared/src/tools/generateCategories.js]: We are indeed using the `csv-parse` module.  But aside from that, we wanted to carefully structure the output.
1. events all go into an array according to their category
2. category names are sorted
3. within a category, events are sorted by documentTitle
4. some commonly-used scenarios were requested to be placed at the top for one category
5. miscellaneous fixing of whitespace in documentTitle strings
CSV parsing itself was straight forward, but structuring the data into something more useful for us was necessary.

[the remark this answers is not in the document]
Comment (waldoj) [reply shared/src/tools/generateCategories.js]: Ah, yes, I see that now. :) Between specifying the delimiter, naming the columns, `whitespaceCleanup`, and the object iterator, my "this is a custom CSV parser" alarm sounded.

File: web-client/src/presenter/computeds/formattedCaseDetail.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (waldoj) [reply web-client/src/presenter/computeds/formattedCaseDetail.js]: Sounds good.

File: README.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
 # Electronic Filing / Case Management System
 
+| Project | Status |
```
[the remark this answers is not in the document]
Comment (kkoskelin) [reply README.md]: We've addressed this by adding a `--no-source-maps` flag when invoking our parcel bundler.  The "Could not load source file" was a warning (not an error, strictly speaking) resulting from a source typescript omission in a module dependency called 'function-tree'.
